types/mp4frag/index.d.ts

```diff
@@ -0,0 +1,183 @@
+// Type definitions for mp4frag 0.6
+// Project: https://github.com/kevinGodell/mp4frag
+// Definitions by: Ken Sanders <https://github.com/kensand>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+import { Transform } from 'stream';
+import 'node/buffer';
+
+export = Mp4Frag;
+
+/**
+ * An SegmentObject.
+ *
+ * An object with values
+ * <code>
+ * {
+ *     segment: null,
+ *     sequence: -1,
+ *     duration: -1;
+ *     timestamp: -1,
+ *     keyframe: true
+ * }
+ * </code
+ * is returned in erroneous conditions (before a segment has been generated, etc. Refer to individual function docs.
+ */
+interface SegmentObject {
+    segment?: Buffer;
+    sequence: number;
+    duration: number;
+    timestamp: number;
+    keyframe: boolean;
+}
+
+interface Mp4FragOptions {
+    /**
+     * Base name of files in m3u8 playlist. Affects the generated m3u8 playlist by naming file fragments.
+     *
+     * Must be set to generate m3u8 playlist. e.g. 'front_door'
+     */
+    hlsPlaylistBase?: string;
+
+    /**
+     * Number of segments to use in m3u8 playlist. Must be an integer ranging from 2 to 20.
+     *
+     * Defaults to 4.
+     */
+    hlsPlaylistSize?: number;
+
+    /**
+     * Number of extra segments to keep in memory. Must be an integer ranging from 0 to 10.
+     *
+     * Defaults to 0.
+     */
+    hlsPlaylistExtra?: number;
+
+    /**
+     * Indicates that m3u8 playlist should be generated after [initialization]{@link Mp4Frag#initialization}
+     * is created and before media segments are created.
+     *
+     * Defaults to true.
+     */
+    hlsPlaylistInit?: boolean;
+
+    /**
+     * Number of segments to keep in memory. Has no effect if using options.hlsPlaylistBase.
+     * Must be an integer ranging from 2 to 30.
+     *
+     * Defaults to 2.
+     */
+    segmentCount?: number;
+}
+
+declare class Mp4Frag extends Transform {
+    /**
+     * Construct a new Mp4Frag object.
+     *
+     * @throws If options.hlsPlaylistBase contains characters other than letters(a-zA-Z) and underscores(_).
+     */
+    constructor(options: Mp4FragOptions);
```

Comment:
aren't options optional?
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L49

Reply:
It's not explicitly said to be optional in the documentation, but from [this line](https://github.com/kevinGodell/mp4frag/blob/master/index.js#L59) it does look to handle a null options field - good catch, will include the change, thank you.